### PR TITLE
Choose right users in federation mode for CopyObject

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Choose right users in federation mode for CopyObject

## Motivation and Context
Choose right users in federation mode for CopyObject

## Regression
No

## How Has This Been Tested?
Using federation setup https://docs.minio.io/docs/minio-federation-quickstart-guide.html and use multi-user feature as well along with it. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

File: cmd/handler-utils.go [hunk 1]
```diff
@@ -176,13 +177,13 @@ func getRedirectPostRawQuery(objInfo ObjectInfo) string {
 	return redirectValues.Encode()
 }
 
-// Returns access key in the request Authorization header.
-func getReqAccessKey(r *http.Request, region string) (accessKey string) {
-	cred, _, _ := getReqAccessKeyV4(r, region)
+// Returns access credentials in the request Authorization header.
+func getReqAccessCred(r *http.Request, region string) (cred auth.Credentials) {
+	cred, _, _ = getReqAccessKeyV4(r, region)
```
Comment (donatello) [inline cmd/handler-utils.go:182]: Is it ok to ignore the error here?

Comment (harshavardhana) [reply cmd/handler-utils.go:182]: yes its okay, since we have already validated the request

**Review by donatello — APPROVED** [review]

**Review by kannappanr — APPROVED** [review]
LGTM. Tested copyobject and copy to another alias using mc.

**Pull request merged**